dotnet/src/Connectors/Connectors.Memory.Milvus/MilvusMemoryStore.cs

```diff
@@ -0,0 +1,560 @@
+        ILoggerFactory? loggerFactory = null)
+        : this(new MilvusClient(host, apiKey, port, ssl, database, callOptions: default, loggerFactory), vectorSize, metricType)
+    {
+        this._ownsMilvusClient = true;
+    }
+
+    /// <summary>
+    /// Initializes a new instance of <see cref="MilvusMemoryStore" /> over the given <see cref="MilvusClient" />.
+    /// </summary>
+    /// <param name="client">A <see cref="MilvusClient" /> configured with the necessary endpoint and authentication information.</param>
+    /// <param name="vectorSize">The size of the vectors used in Milvus. Defaults to 1536.</param>
+    /// <param name="metricType">The metric used to measure similarity between vectors. Defaults to <see cref="SimilarityMetricType.Ip" />.</param>
+    public MilvusMemoryStore(
+        MilvusClient client,
+        int vectorSize = 1536,
+        SimilarityMetricType metricType = SimilarityMetricType.Ip)
+        : this(client, ownsMilvusClient: false, vectorSize, metricType)
+    {
+    }
+
+    private MilvusMemoryStore(
+        MilvusClient client,
+        bool ownsMilvusClient,
+        int vectorSize = 1536,
+        SimilarityMetricType metricType = SimilarityMetricType.Ip)
+    {
+        this.Client = client;
+        this._vectorSize = vectorSize;
+        this._metricType = metricType;
+        this._ownsMilvusClient = ownsMilvusClient;
+    }
+
+    #endregion Constructors
+
+    /// <inheritdoc />
+    public async Task CreateCollectionAsync(string collectionName, CancellationToken cancellationToken = default)
+    {
+        var exists = await this.Client.HasCollectionAsync(collectionName, cancellationToken: cancellationToken).ConfigureAwait(false);
+        if (!exists)
+        {
+            CollectionSchema schema = new()
+            {
+                Fields =
+                {
+                    FieldSchema.CreateVarchar(IdFieldName, maxLength: DefaultVarcharLength, isPrimaryKey: true, autoId: false),
+                    FieldSchema.CreateFloatVector(EmbeddingFieldName, this._vectorSize)
+                },
+                EnableDynamicFields = true
+            };
+
+            MilvusCollection collection = await this.Client.CreateCollectionAsync(collectionName, schema, DefaultConsistencyLevel, cancellationToken: cancellationToken).ConfigureAwait(false);
+
+            await collection.CreateIndexAsync(EmbeddingFieldName, metricType: this._metricType, cancellationToken: cancellationToken).ConfigureAwait(false);
+            await collection.WaitForIndexBuildAsync("float_vector", cancellationToken: cancellationToken).ConfigureAwait(false);
+
+            await collection.LoadAsync(cancellationToken: cancellationToken).ConfigureAwait(false);
+            await collection.WaitForCollectionLoadAsync(waitingInterval: TimeSpan.FromMilliseconds(100), timeout: TimeSpan.FromMinutes(1), cancellationToken: cancellationToken).ConfigureAwait(false);
```

Comment:
Does this hardcoded timeout need to be made configurable in some way?

Reply:
Yeah, we can... There are various potentially configurable Milvus aspects which are currently hard-coded - the consistency level is a good example that's probably more important.

I'm thinking that we use this as a starting point and add configurability in another pass, possibly based on user feedback. I can do that while updating the connector for Milvus 2.3.